hooks/cache_location.py

```diff
@@ -128,12 +128,16 @@ def get_bundle_data_cache_path(self, project_id, plugin_id, pipeline_configurati
         which needs to cache things per-instance can implement this using a sub
         folder inside the bundle cache location).
 
-        :param project_id: The shotgun id of the project to store caches for
+        It is possible to omit some components of the path by explicitly passing
+        a `None` value for them, only the bundle name is required. For example, with
+        `project_id=None`, a site level cache path will be returned.
```

Comment:
I really like this concept that the `site_cache_location` is the same location as the "site project"'s project cache location. I think we should point this out in the docstring - that the *per project* cache location for a site level configuration is the same as the *per site* configuration for any project belonging to that site. I am trying to think if this will cause issues but i don't think it will, as long as we keep things separate.

Reply:
Hum, may be my docstring is confusing? A typical project specific path would be: `<SG site name>/p308c166.basic.maya/<bundle name>` whereas a site path would be `<SG site name>/site/<bundle name>`. Although it is definitely possible to get back something like `<SG site name>/sitec166.basic.maya/<bundle name>` if both the pipe config and plugin ids are given. Is that what you had in mind?